prow/plank/controller.go

```diff
@@ -318,7 +319,9 @@ func (c *Controller) syncJenkinsJob(pj kube.ProwJob, reports chan<- kube.ProwJob
 			pj.Status.State = kube.SuccessState
 			pj.Status.Description = "Jenkins job succeeded."
 			for _, nj := range pj.Spec.RunAfterSuccess {
-				if _, err := c.kc.CreateProwJob(NewProwJob(nj)); err != nil {
+				npj := NewProwJob(nj)
+				npj.Status.ParentName = pj.Metadata.Name
```

Comment:
This would be in the spec, not the status. That way if we rerun it it will keep the same parent.